lib/spack/spack/compilers/clang.py

```diff
@@ -42,18 +42,23 @@ class Clang(Compiler):
     cxx_names = ['clang++']
 
     # Subclasses use possible names of Fortran 77 compiler
-    f77_names = ['gfortran']
+    f77_names = ['flang', 'gfortran']
 
     # Subclasses use possible names of Fortran 90 compiler
-    fc_names = ['gfortran']
+    fc_names = ['flang', 'gfortran']
 
     # Named wrapper links within spack.build_env_path
     link_paths = {'cc': 'clang/clang',
-                  'cxx': 'clang/clang++',
-                  # Use default wrappers for fortran, in case provided in
-                  # compilers.yaml
-                  'f77': 'clang/gfortran',
-                  'fc': 'clang/gfortran'}
+                  'cxx': 'clang/clang++'}
+
+    if sys.platform == 'darwin':
+        # Use default wrappers for fortran, in case provided in
+        # compilers.yaml
+        link_paths['f77'] = 'clang/gfortran'
+        link_paths['fc'] = 'clang/gfortran'
```

Comment:
Does `flang` work on Mac OS X?  And will this break it?  Also, I am guessing people use `gfortran` on Linux with `clang` as well...

This could affect the way `libtool` behaves, though I'm not completely sure it does.  Are there differences in the parameters one passes to `flang` vs. `gfortran` for things like shared libs?  Also, does it hurt to just always call `flang` gfortran? If they're both the same to `libtool`, `gfortran` would be better if it works with older `libtool` versions.   Thoughts?

It might be time to do something we've discussed a bit -- dynamically generate the compiler wrapper env directory for each build.  We've talked about it in the context of supporting Windows -- currently we use static symlinks and we'd need to copy `cc` to all the possible names before each build there b/c Windows has no symlnks.  But this would also help if, as is the case here, the name of the compiler can change from system to system.

Reply:
> Does flang work on Mac OS X?

AFAIK, yes 

> Also, I am guessing people use gfortran on Linux with clang as well...

certainly possible

>  does it hurt to just always call flang gfortran?

i don't know about `flang`, but the other way around (call wrapper `flang` when using `gfortran`) would most likely hurt `openblas`. I remember some long time ago there were issues compiling it on macOS because `clang` Fortran wrappers had  a general name and `openblas` refused to use Fortran compiler.

> dynamically generate the compiler wrapper env directory for each build

that would solve the whole issue and make it work as expected both on macOS and Linux